src/formatDate.js

```diff
@@ -8,7 +8,26 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const separatorFrom = fromFormat[3];
+  const dateParts = date.split(separatorFrom);
+  const dateMap = {};
+
+  for (let i = 0; i < 3; i++) {
+    dateMap[fromFormat[i]] = dateParts[i];
+  }
+
+  if (fromFormat.includes('YYYY') && toFormat.includes('YY')) {
+    dateMap['YY'] = dateMap['YYYY'].slice(2);
+  } else if (fromFormat.includes('YY') && toFormat.includes('YYYY')) {
+    const year = parseInt(dateMap['YY']);
+
+    dateMap['YYYY'] = year < 30 ? '20' + dateMap['YY'] : '19' + dateMap['YY'];
+  }
```

Comment:
The conditionals used to convert between 'YYYY' and 'YY' formats are correct. However, you can optimize the code by avoiding the repetitive access to 'dateMap['YY']' and 'dateMap['YYYY']' by using local variables. This will make the code cleaner and more efficient.